janusgraph-core/src/main/java/org/janusgraph/graphdb/database/StandardJanusGraph.java

```diff
@@ -157,8 +160,12 @@ public StandardJanusGraph(GraphDatabaseConfiguration configuration) {
         //Register instance and ensure uniqueness
         String uniqueInstanceId = configuration.getUniqueGraphId();
         ModifiableConfiguration globalConfig = GraphDatabaseConfiguration.getGlobalSystemConfig(backend);
-        if (globalConfig.has(REGISTRATION_TIME, uniqueInstanceId)) {
+        boolean instanceExists = globalConfig.has(REGISTRATION_TIME, uniqueInstanceId);
+        boolean replaceExistingInstance = configuration.getConfiguration().get(REPLACE_INSTANCE_IF_EXISTS);
+        if (instanceExists && !replaceExistingInstance) {
```

Comment:
based on the config description for `REPLACE_INSTANCE_IF_EXISTS` (_this should be only be a concern..._), you'd need to check that either `UNIQUE_INSTANCE_ID_SUFFIX` or `UNIQUE_INSTANCE_ID` is also set.

Reply:
I think I should remove that sentence from the ConfigurationOption description; I was merely making a testament to the fact that if you let JanusGraph set the id attribute, you will not run into overlapping ids, but if you start meddling with the graph instance id, then we may run into overlapping ids.